### DISCUSSION TITLE
this.supabase.auth.session() returns null

### DISCUSSION THREAD

**Discussion by donkken** [supabase/supabase#2968] [Questions]
Hi, I'm using Supabase together with Ionic Angular. If I build an APK and sign in (using auth) it works, to begin with. If I close my app and leave my device alone for a bit then the authentication returns "null". If I close the app again and start it then I'm signed in and redirected to the correct router. Can't figure out what I'm doing wrong, anyone else experienced anything similar?

Tried it on two diffrent devices and works perfectly if i run it on browser with Ionic serve

**Comment by megacherry** [comment]
I think it's same issue like here https://github.com/supabase/gotrue/issues/173

**Reply by donkken** [reply]
Thanks for the link. Well it is Supabase.auth.onAuthStateChange and Supabase.auth.session() that returns null when it is a returning user. The sign-in method actually works fine

**Reply by donkken** [reply]
```
this.supabase.auth.onAuthStateChange(async (event, session) => {
      console.log('event: ', event);
      if (event === 'SIGNED_IN') {
        this._currentUser.next(session.user);
        this.handleUserDataChange();
      } else if (event === 'USER_UPDATED') {
        this._currentUser.next(session.user);
      } else {
        this._currentUser.next(false);
      }
    });
```

This is the code I'm using. Works fine when I sign in, when I open the app immediately after closure. But fails if I leave it to rest for a while and tries to open it again. Sometimes 10 minutes sometimes 120 minutes.

**Comment by minimaluminium** [comment]
Having the exact same issue with Next.js. This is the code I'm using, and it returns null randomly when the app is loaded for the first time after a while.

```js
useEffect(() => {
  const session = supabase.auth.session()
  console.log(session)
}, [])

useEffect(() => {
  const {data: listener} = supabase.auth.onAuthStateChange(
    (_event, session) => {
      console.log(session)
    }
  )

  return () => {
    listener.unsubscribe()
  }
}, [])
```

**Comment by minimaluminium** [comment]
https://github.com/supabase/gotrue-js/issues/23 seems related.

**Reply by donkken** [reply]
Thanks for the link. Will experiment a bit with it today. I'll let you know if i find a solution.
